source/P0323R9.bs

```diff
@@ -869,11 +872,13 @@ expected<T, E>& operator=(unexpected<G>&& e);
   <tr>
 <td>
 
+  * move constructs an `unexpected<E> tmp` from `move(e.value())` (which may throw)
```

Comment:
Temporary `unexpected<E>` should be created from `unexpected<G>` before destroying val to ensure strong exception guarantee.